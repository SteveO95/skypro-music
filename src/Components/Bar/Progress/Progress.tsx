import React from 'react'
import styles from './Progress.module.css'

type Props = {
	max: number
	value: number
	step: number
	onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
}

export default function Progress({ max, value, step, onChange }: Props) {
	return (
		<input
			className={styles.styledProgressInput} // Применение стилей к ползунку
			type='range' // Тип элемента - ползунок
			min='0' // Минимальное значение ползунка
			max={max} // Максимальное значение, зависит от длительности аудио
			value={value} // Текущее значение ползунка
			step={step} // Шаг изменения значения
			onChange={onChange} // Обработчик события изменения
		/>
	)
}
