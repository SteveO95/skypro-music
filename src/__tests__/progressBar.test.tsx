import Progress from '@/components/Bar/Progress/Progress'
import renderer from 'react-test-renderer'

describe('Индикатор прогресса', () => {
	it('рендер прогресс бара', () => {
		const props = {
			max: 100,
			value: 10,
			step: 0.5,
			onChange: () => {},
		}

		const bar = renderer
			.create(
				<Progress
					max={props.max}
					value={props.value}
					step={props.step}
					onChange={props.onChange}
				/>,
			)
			.toJSON()
		expect(bar).toMatchSnapshot()
	})
})
