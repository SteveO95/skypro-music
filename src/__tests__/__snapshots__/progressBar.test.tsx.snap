// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Индикатор прогресса рендер прогресс бара 1`] = `
<input
  className="styledProgressInput"
  max={100}
  min="0"
  onChange={[Function]}
  step={0.5}
  type="range"
  value={10}
/>
`;
